contracts/service-registry/src/contract/execute.rs

```diff
@@ -78,6 +79,33 @@ pub fn update_verifier_authorization_status(
     Ok(Response::new())
 }
 
+pub fn update_service(
+    deps: DepsMut,
+    service_name: String,
+    updated_service_params: UpdatedServiceParams,
+) -> Result<Response, ContractError> {
+    SERVICES.update(deps.storage, &service_name, |service| match service {
+        None => Err(ContractError::ServiceNotFound),
+        // TODO: write a macro to automate this?
```

Comment:
Unless we go all out and create a derive macro like `UpdatableParams` that automatically generates the Params struct generically and the update function with it the effort of creating a macro is not worth it. So I suggest to check how often we would need that and then either commit to doing that right now (follow-up PR), or remove this TODO because it's not worth it